src/runtime/opengl/opengl_module.h

```diff
@@ -0,0 +1,89 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file opengl_module.h
+ * \brief Execution handling of OpenGL kernels
+ */
+#ifndef TVM_RUNTIME_OPENGL_OPENGL_MODULE_H_
+#define TVM_RUNTIME_OPENGL_OPENGL_MODULE_H_
+
+#include <tvm/runtime/config.h>
+#include <tvm/runtime/packed_func.h>
+#include <memory>
+#include <vector>
+#include <string>
+#include "../meta_data.h"
+
+namespace tvm {
+namespace runtime {
+
+enum OpenGLArgKind : int {
+  kInputTexture = 0,
+  kOutputTexture = 1,
+  kUniform = 2,
+};
+
+struct OpenGLShader {
+  std::string source;
+  std::vector<std::string> arg_names;
```

Comment:
document the arg_kinds, refer to OpenGLArgKind

Reply:
I did something different: when outputing to JSON, transform to string. This gives better readability.